### DISCUSSION TITLE
Drive cycle simulation fails after several set of cycles

### DISCUSSION THREAD

**Discussion by manjunathnilugal** [pybamm-team/PyBaMM#3025] [Q&A]
Hello everyone, 

I am trying the drive simulation for 1000 cycles, I have given all the functions that are defined f(x) within the bounds [0, 1]. The drive cycle simulation runs for a particular cycles and then throws an error as mentioned below.

`2023-06-11 01:08:01.299 - [ERROR] callbacks.on_experiment_error(226): Simulation error: Solver failed because the following interpolation bounds were exceeded: ["Interpolant 'interpolating_function' upper bound"]. You may need to provide additional interpolation points outside these bounds.`

The issue is in continuation with the similar issue  #2714 
Kindly let me know if there is nay method to resolve this issue.

![image](https://github.com/pybamm-team/PyBaMM/assets/82285213/2f007bfd-f691-4dc6-9ab9-afecaea38f69)

**Comment by brosaplanella** [comment]
Can you please share a minimum working example so we can reproduce the error on our end?
